cmd/analyze-xhprof.go

```diff
@@ -14,15 +12,17 @@ func init() {
 	RootCmd.AddCommand(xhprofCmd)
 	xhprofCmd.Flags().StringVarP(&field, "field", "f", "excl_wt", "Field to view/sort (wt, excl_wt, cpu, excl_cpu, memory, excl_memory, io, excl_io)")
 	xhprofCmd.Flags().Float32VarP(&minPercent, "min", "m", 1, "Display items having minimum percentage (default 1%) of --field, with respect to main()")
+	xhprofCmd.Flags().BoolVarP(&callgrind, "callgrind", "c", false, "Filepaths are in callgrind format (e.g. generated by xdebug)")
```

Comment:
Is it possible to parse the file as xhprof format, then when it fails (because its not json) try again with callgrind format? This way you wouldn't need to pass the flag. although it might be a bit weird to have the command named `analyze-xhprof` in this case.

Reply:
Another point is that when providing a flag, only callgrind files will be accepted, and there won't be mixing with xhprof format, which i think should be the case, as there are some differences (e.g. callgrind has no mem/io, it includes `require` and `include` as functions, etc.)

Reply:
@s1na in this case, please add two more commands "analyze-callgrind" and "compare-callgrind" instead of making it a flag.